### PR TITLE
[Feature] Support QDTrack

### PR DESCRIPTION
**Pull request by JingweiZhang12**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/losses/l2_loss.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import torch
+from mmengine.model import BaseModule
+from torch import Tensor
+
+from mmdet.registry import MODELS
+from .utils import weighted_loss
+
+
+@weighted_loss
+def l2_loss(pred: Tensor, target: Tensor) -> Tensor:
```
Comment (RangiLyu) [inline mmdet/models/losses/l2_loss.py:14]: We have MSELoss.

File: mmdet/models/losses/l2_loss.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        pred = torch.clamp(pred, min=0, max=1)
+
+        num_pos = int((target == 1).sum())
+        num_neg = int((target == 0).sum())
+        if self.neg_pos_ub > 0 and num_neg / (num_pos +
+                                              1e-6) > self.neg_pos_ub:
+            num_neg = num_pos * self.neg_pos_ub
+            neg_idx = torch.nonzero(target == 0, as_tuple=False)
+
+            if self.hard_mining:
```
Comment (RangiLyu) [inline mmdet/models/losses/l2_loss.py:106]: The sampling process of mmdet is placed in the sampler of the head, not in loss.

File: mmdet/models/losses/multipos_cross_entropy_loss.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            label (torch.Tensor): The assigned label of the prediction.
+            weight (torch.Tensor): The element-wise weight.
+            reduction (str): Same as built-in losses of PyTorch.
+            avg_factor (float): Average factor when computing
+                the mean of losses.
+        Returns:
+            torch.Tensor: Calculated loss
+        """
+
+        pos_inds = (label >= 1)
```
Comment (RangiLyu) [inline mmdet/models/losses/multipos_cross_entropy_loss.py:39]: what is the dtype of label? long or float?

**Comment by JingweiZhang12** [comment]
The PR waits for the merge of evaluation PR.

File: mmdet/models/losses/l2_loss.py (hunk 2 above)
Comment (JingweiZhang12) [reply mmdet/models/losses/l2_loss.py:126]: The hard mining here (in ReID) is different from the sampling process in detection.

File: mmdet/models/losses/l2_loss.py (hunk 1 above)
Comment (JingweiZhang12) [reply mmdet/models/losses/l2_loss.py:14]: I  rename this file as `margin_loss.py`

File: mmdet/models/losses/multipos_cross_entropy_loss.py (hunk 3 above)
Comment (JingweiZhang12) [reply mmdet/models/losses/multipos_cross_entropy_loss.py:47]: long

**Review by hhaAndroid — APPROVED** [review]
